backend/amp-live-list.go

```diff
@@ -86,25 +119,26 @@ func readStatus(r *http.Request) int {
 	return result
 }
 
-func createPage(newStatus int) Page {
+func createPage(newStatus int, timestamp time.Time) Page {
 	if newStatus > len(blogs) {
 		newStatus = len(blogs)
 	}
-	blogItems := getBlogEntries(newStatus)
+	blogItems := getBlogEntries(newStatus, timestamp)
 	score := createScore(newStatus, 0)
 	return Page{BlogItems: blogItems, FootballScore: score}
 }
 
-func renderAmpLiveListSample(w http.ResponseWriter, page Page) {
-	t, _ := template.ParseFiles("dist/components/amp-live-list/index.html")
+func renderSample(w http.ResponseWriter, r *http.Request, filePath string) {
+	t, _ := template.ParseFiles(filePath)
 	w.Header().Set("Cache-Control", fmt.Sprintf("max-age=%d, public, must-revalidate", MAX_AGE_IN_SECONDS))
-	t.Execute(w, page)
+	newStatus := updateStatus(w, r)
+	t.Execute(w, createPage(newStatus, time.Now()))
 }
 
-func getBlogEntries(size int) []BlogItem {
+func getBlogEntries(size int, timestamp time.Time) []BlogItem {
 	result := make([]BlogItem, 0)
 	for i := 0; i < size; i++ {
-		result = append(result, blogs[i].clone())
+		result = append(result, blogs[i].cloneWith(i+1, timestamp.Add(time.Duration(FIFTEEN_SECONDS*(size-i))*time.Second)))
```

Comment:
FIFTEEN_SECONDS is a negative number. While Add accepts a duration, Sub requires a time.[API](https://golang.org/pkg/time/)